src/ds/dllist.h

```diff
@@ -6,16 +6,24 @@
 
 namespace snmalloc
 {
-  template<class T, uintptr_t terminator = 0>
+  template<class T, uintptr_t terminator_ = 0>
```

Comment:
Why is this getting a trailing underscore?  It looks like a typo everywhere.  Isn't the convention to use Lead Capitals (or ALL CAPITALS) for template parameters to differentiate them from other values?